src/utils/cutPlane.js

```diff
@@ -0,0 +1,203 @@
+import {Box3, BufferAttribute, BufferGeometry, Mesh, Vector3} from 'three'
+
+
+/* eslint-disable no-magic-numbers */
+/**
+ * getSelectionAxisFromBoundingBox is the helper method for the cutplane logic
+ *
+ * @param {Object} boundingBox bouding box
+ * @return {Object}
+ */
+export function getSelectionAxisFromBoundingBox(boundingBox) {
+  return {
+    x: {
+      size: Math.abs( boundingBox.max.x - boundingBox.min.x),
+      center: (boundingBox.max.x + boundingBox.min.x) / 2,
+    },
+    y: {
+      size: Math.abs( boundingBox.max.y - boundingBox.min.y),
+      center: (boundingBox.max.y + boundingBox.min.y) / 2,
+    },
+    z: {
+      size: Math.abs(boundingBox.max.z - boundingBox.min.z ),
+      center: (boundingBox.max.z + boundingBox.min.z) / 2,
+    },
+  }
+}
+
+
+/**
+ * getModelCenter is the helper method for the cutplane logic
+ *
+ * @param {Object} ifcModel bouding box
+ * @return {Object} centerCoordinates
+ */
+/**
+ * getModelCenter is the helper method for the cutplane logic
+ *
+ * @param {Object} ifcModel bouding box
+ * @return {Object} centerCoordinates
+ */
+export function getModelCenter(ifcModel) {
```

Comment:
Hey Oleg, if you change the return of this function to new Vec3, the section plane should work:

```
export function getModelCenter(ifcModel) {
  return new Vector3(
    (ifcModel?.geometry.boundingBox.max.x +
      ifcModel?.geometry.boundingBox.min.x) /
    2,
    (ifcModel?.geometry.boundingBox.max.y +
      ifcModel?.geometry.boundingBox.min.y) /
    2,
    (ifcModel?.geometry.boundingBox.max.z +
      ifcModel?.geometry.boundingBox.min.z) /
    2,
  )
}
```